src/Compilers/CSharp/Portable/Generated/CSharp.Generated.g4

```diff
@@ -335,6 +336,14 @@ array_rank_specifier
   : '[' (expression (',' expression)*)? ']'
   ;
 
+function_pointer_type
+  : 'delegate' '*' ('cdecl' | 'managed' | 'stdcall' | 'thiscall' | 'unmanaged')? '<' function_pointer_parameter_or_return_type (',' function_pointer_parameter_or_return_type)* '>'
+  ;
+
+function_pointer_parameter_or_return_type
```

Comment:
maybe just `ModifiedType`.

Reply:
I wanted to be very specific with the grammar here because this can accept both return-type modifiers, like `ref readonly`, and parameter-type modifiers, like `in`.

---
In reply to: [348012458](https://github.com/dotnet/roslyn/pull/39879#discussion_r348012458) [](ancestors = 348012458)

Reply:
I don't see the problem.  `ParameterSyntax` (and `ModifiedSyntax` if you went that route) would allow this.  My point is simply not to call it FunctionPointerParameterOrReturnType.  It's really unnecessarily specific.  It's just a type with modifiers.  Frankly, a reasonable way to represent it IMO would be like so:

```g4
function_pointer_type
  : 'delegate' '*' ('cdecl' | 'managed' | 'stdcall' | 'thiscall' | 'unmanaged')? '<' type (',' type)* '>'

type
   : /* ... */
   : modified_type;

modified_type
    : modifier* type;
```

Reply:
Well, the point is that `ParameterSyntax` technically allows this, from a "this will only error in semantic binding" perspective, but in this grammar at least, that would be an error.

---
In reply to: [348212328](https://github.com/dotnet/roslyn/pull/39879#discussion_r348212328) [](ancestors = 348212328)

Reply:
sorry, i feel like i'm not understanding the point you're making.  want to try gitter/slack?